src/main/java/org/broadinstitute/dsde/consent/ontology/translate/utils/SearchUtils.java

```diff
@@ -0,0 +1,18 @@
+package org.broadinstitute.dsde.consent.ontology.translate.utils;
+
+import org.broadinstitute.dsde.consent.ontology.Utils;
+
+import java.util.regex.Pattern;
+
+public class SearchUtils {
+    public static boolean searchForKeyword(final String keyword, final String targetText) {
+        if (Utils.isNullOrEmpty(keyword) || Utils.isNullOrEmpty(targetText)) {
+            return false;
+        }
+
+        return Pattern
+                .compile(keyword, Pattern.CASE_INSENSITIVE)
+                .matcher(targetText)
+                .find();
```

Comment:
Unless `keyword` contains a regular expression (or using `Pattern` is more performant), you can do a substring search here. Using apache commons StringUtils handles the null checks, unless you want to return `false` for a substring search for the empty string, which would normally return `true`.
```
    return StringUtils.containsIgnoreCase(targetText, keyword);
```

Reply:
That sounds preferable.